### PR TITLE
MurckoScaffold Fix

### PR DESCRIPTION
**Pull request by skrsna**
*Issue #, if available:*

Addresses #84 

*Description of changes:*

Replaced `MurckoScaffoldSmiles` with `AllChem.MurckoDecompose` and deleted `includeChirality` arg in `ScaffoldSplitter.train_val_test_split`, `k_fold_split` and `test_splitters.py` 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by skrsna** [comment]
Seems like the PR is failing because of linting. Is there a shell script to format the code?

File: python/dgllife/utils/splitters.py [hunk 1]
```diff
@@ -484,23 +485,26 @@ def get_ordered_scaffold_sets(molecules, include_chirality, log_every_n):
...
                 FastFindRings(mol)
-                mol_scaffold = MurckoScaffold.MurckoScaffoldSmiles(
-                    mol=mol, includeChirality=include_chirality)
+                mol_scaffold = AllChem.MurckoDecompose(mol)
                 # Group molecules that have the same scaffold
-                scaffolds[mol_scaffold].append(i)
+                if mol_scaffold not in scaffolds:
+                    scaffolds[mol_scaffold] = [i]
+                else:
+                    scaffolds[mol_scaffold].append(i)
```
Comment (mufeili) [inline python/dgllife/utils/splitters.py:500]: Any reasons for preferring this over `defaultdict(list)`?

Comment (skrsna) [reply python/dgllife/utils/splitters.py:500]: My bad, I added this for debugging. You can change it back to defaultdict

File: python/dgllife/utils/splitters.py [hunk 2]
```diff
@@ -574,7 +577,8 @@ def train_val_test_split(dataset, mols=None, sanitize=True, include_chirality=Fa
                     val_indices.extend(group_indices)
             else:
                 train_indices.extend(group_indices)
-
+        assert len(set(train_indices).intersection(set(val_indices))) == 0, "Failed to successfully split using scaffolds"
+        assert len(set(test_indices).intersection(set(val_indices))) == 0, "Failed to successfully split using scaffolds"
```
Comment (mufeili) [inline python/dgllife/utils/splitters.py:581]: Why did you add the two assertions here? L572-L579 should exclude the possibility of overlappings.

**Comment by mufeili** [comment]
> Seems like the PR is failing because of linting. Is there a shell script to format the code?

You can simply click "Details" to check what went wrong for CI. For this PR, the lint check fails with the following error messages:

```
python/dgllife/utils/splitters.py:580:0: C0301: Line too long (122/100) (line-too-long)

python/dgllife/utils/splitters.py:581:0: C0301: Line too long (121/100) (line-too-long)

python/dgllife/utils/splitters.py:588:21: W0613: Unused argument 'include_chirality' (unused-argument)

python/dgllife/utils/splitters.py:12:0: W0611: Unused defaultdict imported from collections (unused-import)

python/dgllife/utils/splitters.py:18:0: W0611: Unused MurckoScaffold imported from rdkit.Chem.Scaffolds (unused-import)
```

File: python/dgllife/utils/splitters.py (hunk 2 above)
Comment (skrsna) [reply python/dgllife/utils/splitters.py:581]: I had these for debugging. I'll delete them and push the change. Thanks!!

**Comment by mufeili** [comment]
I think you also need to remove:
- the description of the `include_chirality` parameter in L537-538
- the argument `include_chirality` at L582

**Comment by skrsna** [comment]
I'm slightly confused with pylint. I ran the `task_lint.sh` locally before pushing and I didn't see `bad continuation` error. But the CI is crashing with that error on Jenkins. On the other hand when I run `task_lint.sh` locally I get the following messages which I think are supposed to be ignored by the pylintrc. Sorry about this! 

```

Messages
--------

+---------------------+------------+
|message id           |occurrences |
+=====================+============+
|super-with-arguments |67          |
+---------------------+------------+
|abstract-method      |65          |
+---------------------+------------+
|no-member            |9           |
+---------------------+------------+
|not-callable         |1           |
+---------------------+------------+
```

**Comment by mufeili** [comment]
That's a bit strange. Let me merge this PR first and do some minor adjustment afterwards. Thank you for your contribution!
